README.md

```diff
@@ -187,6 +187,57 @@ for grabber in grabbers.values():
     grabber.release()
 ```
 
+#### RTSP Discovery
+RTSP cameras with support for ONVIF can be discovered using the tool built into the library. It provides a convenient method for finding all the RTSP cameras in the local network.
+
+```python
+from framegrab import RTSPDiscovery, ONVIFDeviceInfo
+        
+devices = RTSPDiscovery.discover_camera_ips()
+```
+
+The `discover_camera_ips()` will provide a list of devices that it finds in the `ONVIFDeviceInfo` format. An optional mode `auto_discover_modes` can be used to try different default credentials to fetch RTSP URLs:
+
+light: only try first three usernames and passwords ("admin:admin", "admin:", and no username/password).
+
+complete_fast: try the entire DEFAULT_CREDENTIALS without delays in between. 
+
+complete_slow: try the entire DEFAULT_CREDENTIALS with a delay of 1 seconds in between.
+
+Defaults to None.
+
+```python
+from pydantic import BaseModel
+# Model for storing ONVIF Device Information
+class ONVIFDeviceInfo(BaseModel):
+    ip: str
+    port: Optional[int] = 80
+    username: Optional[str] = ""
+    password: Optional[str] = ""
+    xaddr: Optional[str] = ""  # ONVIF service address
+    rtsp_urls: Optional[List[str]] = []  # List of rtsp urls for the camera
+```
+
+After getting the list and enter the username and password of the camera. Use `generate_rtsp_urls()` to generate RTSP URLs for each devices.
```

Comment:
I wonder if this should be accessible as a parameter of `discover_camera_ips`. For example you could do: 
```
RTSPDiscovery.discover_camera_ips(autodiscover_mode='light', generate_rtsp_urls=True)
```

In this way, you can just run one line of code to take care of everything, which I believe is what users will want to do.

Reply:
The code will automatically call `_try_logins` which then calls `generate_rtsp_urls` if the `autodiscovermode` is set to anything other than none. 

Therefore, it is essentially doing everything in one function, although sometimes it is better for the user to manually enter the username and password because you might get locked out from the cameras if there are too many incorrect attempts.